### ISSUE TITLE
Bug: `object-shorthand` rule don't report problems with `["error", 'always']` option

### ISSUE THREAD

**Issue by B2o5T** [eslint/eslint#15593]
### Environment

Node version: 
npm version: 
Local ESLint version: 8.8.0
Global ESLint version: 
Operating System: 

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint object-shorthand: ["error", 'always']*/
/*eslint-env es6*/

var foo = {
  foo: (bar, baz) => {
    return bar + baz;
  },

  qux: (foobar) => {
    return foobar * 2;
  }
};
```
https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgb2JqZWN0LXNob3J0aGFuZDogW1wiZXJyb3JcIiwgJ2Fsd2F5cyddKi9cbi8qZXNsaW50LWVudiBlczYqL1xuXG52YXIgZm9vID0ge1xuICBmb286IChiYXIsIGJheikgPT4ge1xuICAgIHJldHVybiBiYXIgKyBiYXo7XG4gIH0sXG5cbiAgcXV4OiAoZm9vYmFyKSA9PiB7XG4gICAgcmV0dXJuIGZvb2JhciAqIDI7XG4gIH1cbn07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjYXBpdGFsaXplZC1jb21tZW50cyI6Mn0sImVudiI6e319fQ==

### What did you expect to happen?

receive 2 errors

### What actually happened?

everything is fine 😩

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

`/* eslint object-shorthand: "error" */` directive comment don't work also

**Comment by mdjermanovic** [comment]
Hi @B2o5T, thanks for the issue!

Arrow functions are allowed by default. You can set [`{ avoidExplicitReturnArrows: true }`](https://eslint.org/docs/rules/object-shorthand#avoidexplicitreturnarrows) to disallow arrow functions used like methods:

```js
/*eslint object-shorthand: ["error", "always", { "avoidExplicitReturnArrows": true }] */
/*eslint-env es6*/

var foo = {
  foo: (bar, baz) => { // error
    return bar + baz;
  },

  qux: (foobar) => { // error
    return foobar * 2;
  }
};
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgb2JqZWN0LXNob3J0aGFuZDogW1wiZXJyb3JcIiwgXCJhbHdheXNcIiwgeyBcImF2b2lkRXhwbGljaXRSZXR1cm5BcnJvd3NcIjogdHJ1ZSB9XSAqL1xuLyplc2xpbnQtZW52IGVzNiovXG5cbnZhciBmb28gPSB7XG4gIGZvbzogKGJhciwgYmF6KSA9PiB7IC8vIGVycm9yXG4gICAgcmV0dXJuIGJhciArIGJhejtcbiAgfSxcblxuICBxdXg6IChmb29iYXIpID0+IHsgLy8gZXJyb3JcbiAgICByZXR1cm4gZm9vYmFyICogMjtcbiAgfVxufTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

I'm closing this issue as it seems that everything works as intended.
